src/download-manager.js

```diff
@@ -37,205 +38,102 @@ const browserManager = require('./browser-manager.js');
-                });
-              }
-            });
-            break;
-          case 'darwin':
-            releaseAssets.forEach(download => {
-              if (download.name.indexOf('mac') !== -1) {
-                return resolve({
-                  url: download.browser_download_url,
-                  name: download.name
-                });
-              }
-            });
-            break;
-          case 'windows':
-            releaseAssets.forEach(download => {
-              if (download.name.indexOf('win64') !== -1) {
-                return resolve({
-                  url: download.browser_download_url,
-                  name: download.name
-                });
-              }
-            });
-            break;
-          default:
-            return reject(new Error('Unsupported platform: ' +
-              process.platform));
-        }
-
-        return reject(new Error('Unable to find appropriate download for ' +
-          'this environment.'));
-      });
+  _getStorage() {
+    storage.initSync({
+      dir: path.join(application.getInstallDirectory(), 'database/')
     });
+
+    return storage;
   }
 
   /**
-   * This method retrieves the Firefox Gecko driver, and makes it
-   * executable. It's installed to the directory defined via the
-   * application get/setInstallDir() and it's added to the processes
-   * PATH for use by selenium-webdriver.
-   * @return {Promise}              Resolves when the driver is downloaded
+   * This method will download a browser if it is needed (i.e. can't be found
+   * in the usual system location or in the install directory).
+   * @param  {String} browserId This is the Selenium ID of the browser you wish
+   *                            to download ('chrome', 'firefox', 'opera').
+   * @param  {String} release   This downloads the browser on a particular track
+   *                            and can be 'stable', 'beta' or 'unstable'
+   * @param  {int} [expirationInHours=24] This is how long until a browser
+   *                             download is regarded and expired and Should
+   *                             be updated. A value of 0 will force a download.
+   *         										 If you want to install the browser regardless
+   *                             of any existing installs of the process, pass
+   *                             in true.
+   * @return {Promise}           Promise resolves once the browser has been
+   *                             downloaded and ready for use.
    */
-  downloadFirefoxDriver() {
-    let geckoDriverPath = path.join(
-      application.getInstallDirectory(), 'geckodriver');
+  downloadBrowser(browserId, release, expirationInHours) {
+    let installDir = application.getInstallDirectory();
+
+    const storage = this._getStorage();
+    const storageKey = `${browserId}:${release}`;
 
     return new Promise((resolve, reject) => {
-      mkdirp(geckoDriverPath, err => {
+      storage.getItem(storageKey, (err, value) => {
```

Comment:
Not a fan of adding a library to do this.

I had considered moving to seperate model to hide this away but given it's limited usage I don't think it's too bad at the moment.